lua/cmp/source.lua

```diff
@@ -115,7 +115,10 @@ source.get_entries = function(self, ctx)
     if e.score >= 1 then
       e.matches = match.matches
       e.exact = e:get_filter_text() == inputs[o] or e:get_word() == inputs[o]
-      table.insert(entries, e)
+
+      if self:get_entry_filter()(e, ctx) then
```

Comment:
Move this out of the loop, fetching the filter for each iteration seems a tad unnecessary.